packages/@ngtools/benchmark/src/benchmark.ts

```diff
@@ -0,0 +1,113 @@
+import * as fs from 'fs';
+
+import { BenchmarkOptions } from './benchmark-options';
+import {
+  benchmarkLog,
+  isNumber,
+  average,
+  combine,
+  makeArray,
+  makeMatchFn,
+  matchSpawn,
+  serialMultiPromise
+} from './utils';
+
+export const benchmark = (benchmarkOptions: BenchmarkOptions) => {
+  const log = benchmarkLog.bind(null, benchmarkOptions);
+  const logIfDefined = (str: string, prop: any) => {
+    if (Array.isArray(prop) && prop.length === 0) {
+      prop = null;
+    }
+    return prop ? log(str + prop) : null;
+  };
+
+  const cwd = process.cwd();
+
+  // Build match function
+  let matchFn: any = null;
+  if (benchmarkOptions.match) {
+    matchFn = makeMatchFn(
+      benchmarkOptions.debug,
+      benchmarkOptions.match,
+      benchmarkOptions.matchCount,
+      benchmarkOptions.matchEditFile,
+      benchmarkOptions.matchEditString,
+    );
+  }
+
+  let editedFileContents: string;
+  if (benchmarkOptions.matchEditFile) {
+    // backup contents of file that is being edited for rebuilds
+    editedFileContents = fs.readFileSync(benchmarkOptions.matchEditFile, 'utf8');
+  }
+  // combine flags commands
+  let flagCombinations = combine(benchmarkOptions.extraArgs);
+  flagCombinations.unshift([]);
+
+  const startTime = Date.now();
+  log(`Base command: ${benchmarkOptions.command}`);
```

Comment:
Order wouldn't be guaranteed with those, nor would they allow to show a more verbose text as easily.